pkg/operation/botanist/component/extensions/operatingsystemconfig/operatingsystemconfig.go

```diff
@@ -497,7 +499,7 @@ func (o *operatingSystemConfig) newDeployer(osc *extensionsv1alpha1.OperatingSys
 		clusterDomain:           o.values.ClusterDomain,
 		criName:                 criName,
 		images:                  o.values.Images,
-		kubeletCACertificate:    o.values.KubeletCACertificate,
+		kubeletCABundle:         string(kubeletCASecret.Data[secretutils.DataKeyCertificateBundle]),
```

Comment:
You could change `kubeletCABundle` to type `[]byte` (later on it's converted from `string` to `[]byte` anyways)